step-certificates/templates/secrets.yaml

```diff
@@ -39,13 +39,23 @@ type: smallstep.com/private-keys
 metadata:
   name: {{ include "step-certificates.fullname" . }}-secrets
   namespace: {{ .Release.Namespace }}
-stringData:
-  intermediate_ca_key: |-
-    {{- .Values.inject.secrets.x509.intermediate_ca_key  | nindent 4 }}
-  root_ca_key: |-
-    {{- .Values.inject.secrets.x509.root_ca_key | nindent 4 }}
-  ssh_host_ca_key: |-
-    {{- .Values.inject.secrets.ssh.host_ca_key | nindent 4 }}
-  ssh_user_ca_key: |-
-    {{- .Values.inject.secrets.ssh.user_ca_key | nindent 4 }}
+data:
+  intermediate_ca_key: {{ .Values.inject.secrets.x509.intermediate_ca_key }}
+  root_ca_key: {{ .Values.inject.secrets.x509.root_ca_key }}
+  ssh_host_ca_key: {{ .Values.inject.secrets.ssh.host_ca_key }}
+  ssh_user_ca_key: {{ .Values.inject.secrets.ssh.user_ca_key }}
 {{- end }}
+---
+{{- if .Values.inject.enabled }}
+apiVersion: v1
+kind: Secret
+type: smallstep.com/certs
+metadata:
+  name: {{ include "step-certificates.fullname" . }}-certs
```

Comment:
I agree when we say that a cert is public, the point is that probably many people can use a helm chart in a CI/CD environment, so they will have to substitute variables in the values.yaml and here it comes the problem. 

A certificate or a key, because of their structure and length are difficult to store in an environment variable to be replaced later, if you try it  you will get an error. The only way to do it is to save the certificate in oneline and then do tricks and things to have it the same way. Instead if you save your certificate or key in base64 in oneline it will always preserves the structure and it is easier then to replace it.

Reply:
@ricosega Moving the ConfigMaps to secrets might be a breaking change for some users. Can you investigate if storing the data as `binaryData` in the ConfigMap will solve your use case?
See https://kubernetes.io/docs/concepts/configuration/configmap/#configmap-object

PS: I've been on vacation.

Reply:
Sorry for the delay I've been on vacation too. 
If we do not want to create a breaking change to any current user the only thing we can do is to make it retrocompatible, keeping the previous config and adding this new one for example with a value like, **isBase64Encoded = bool**.
What do you think?

Reply:
@ricosega I think its best not to introduce a breaking change for this.  I would be open to seeing a `isBase64Encoded` option that allows the user to pass their value base64 encoded with the value of said flag defaulting to `false`.  The helm chart could then decode the value and place it in the configmap.

Also, you may want to update your branch to the latest master.  Support for registration authorities was added while you were away and the latest version of the chart is now `1.16.1`